File: binding/src/libqlatin1stringmatcher.cpp
#include <QLatin1StringMatcher>
#include <qlatin1stringmatcher.h>
#include "libqlatin1stringmatcher.h"
#include "libqlatin1stringmatcher.hxx"

QLatin1StringMatcher* QLatin1StringMatcher_new() {
	 return new QLatin1StringMatcher();
}

void QLatin1StringMatcher_SetCaseSensitivity(QLatin1StringMatcher* self, int cs) {
	self->setCaseSensitivity(static_cast<Qt::CaseSensitivity>(cs));
}

int QLatin1StringMatcher_CaseSensitivity(const QLatin1StringMatcher* self) {
	return self->caseSensitivity();
}

void QLatin1StringMatcher_Delete(QLatin1StringMatcher* self) {
    delete self;
}

File: binding/src/libqlatin1stringmatcher.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qlatin1stringmatcher.html
pub const qlatin1stringmatcher = struct {

    /// New constructs a new QLatin1StringMatcher object.
    pub fn New() QtC.QLatin1StringMatcher {
        return qtc.QLatin1StringMatcher_new();
    }


    pub fn SetCaseSensitivity(self: ?*anyopaque, cs: i32) void {
        qtc.QLatin1StringMatcher_SetCaseSensitivity(@ptrCast(self), @intCast(cs));
    }

    pub fn CaseSensitivity(self: ?*anyopaque, ) i32 {
        return qtc.QLatin1StringMatcher_CaseSensitivity(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QLatin1StringMatcher_Delete(@ptrCast(self));
    }
};
